include/storage/shared_datatype.hpp

```diff
@@ -171,6 +172,20 @@ struct SharedDataLayout
 
         return ptr;
     }
+    friend std::ostream &operator<<(std::ostream &os, const DataLayout &layout)
```

Comment:
Could we try to keep the debugging code separate from the implementation? @MoKob did something very useful like this by adding a `debug.hpp` header. This makes it a lot easier to spot debug code in a code review.